src/test/java/com/iota/iri/service/tipselection/impl/EntryPointSelectorCumulativeWeightThresholdTest.java

```diff
@@ -183,27 +195,34 @@ public void succeedWhenEntryPointSizeIsJustRight() throws Exception {
         final int threshold = 10;
         final int longChainLength = (int)(EntryPointSelectorCumulativeWeightThreshold.MAX_SUBTANGLE_SIZE / 2.5);
         final int shortChainLength = 2;
-        Hash longChainTip = Hash.NULL_HASH;
-        Hash shortChainTip= Hash.NULL_HASH;
-
-        for (int i = 0; i < longChainLength; i++) {
-            TransactionViewModel newTip = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(longChainTip, longChainTip), getRandomTransactionHash());
-            newTip.store(tangle);
-            longChainTip = newTip.getHash();
-        }
-
-        for (int i = 0; i < shortChainLength; i++) {
-            TransactionViewModel newTip = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(shortChainTip, shortChainTip), getRandomTransactionHash());
-            newTip.store(tangle);
-            shortChainTip = newTip.getHash();
-        }
+        
+        makeChain(longChainLength);
+        List<Hash> shortChain = makeChain(shortChainLength);
 
-        Mockito.when(tipsViewModel.getRandomSolidTipHash()).thenReturn(shortChainTip);
+        // Start from the short chain, so we start the BFS from the genesis
+        Mockito.when(tipsViewModel.getRandomSolidTipHash()).thenReturn(shortChain.get(shortChain.size() - 1));
         
-        EntryPointSelector entryPointSelector = new EntryPointSelectorCumulativeWeightThreshold(tangle, tipsViewModel, threshold, walker, walkValidator);
+        EntryPointSelector entryPointSelector = new EntryPointSelectorCumulativeWeightThreshold(tangle, tipsViewModel, threshold);
 
         Hash entryPoint = entryPointSelector.getEntryPoint();
 
         Assert.assertEquals(Hash.NULL_HASH, entryPoint);
     }
+
+    private List<Hash> makeChain(int length) throws Exception {

```

Comment:
would be nice to have the ability to add a starting point, so you could break off in different points as well.